src/btree/bt_cursor.c

```diff
@@ -470,6 +470,45 @@ __wt_btcur_reset(WT_CURSOR_BTREE *cbt)
 	return (__cursor_reset(cbt));
 }
 
+/*
+ * __wt_btcur_search_uncommitted --
+ *	Search for a matching uncommitted record in the tree.
+ */
+int
+__wt_btcur_search_uncommitted(WT_CURSOR_BTREE *cbt, WT_UPDATE **updp)
+{
+	WT_BTREE *btree;
+	WT_CURSOR *cursor;
+	WT_DECL_RET;
+	WT_SESSION_IMPL *session;
+	WT_UPDATE *upd;
+
+	btree = cbt->btree;
+	cursor = &cbt->iface;
+	session = (WT_SESSION_IMPL *)cursor->session;
+	upd = NULL;					/* -Wuninitialized */
+
+	WT_ERR(btree->type == BTREE_ROW ?
+	    __cursor_row_search(session, cbt, NULL, false) :
+	    __cursor_col_search(session, cbt, NULL));
+
+	WT_ASSERT(session, cbt->compare == 0);
```

Comment:
This assert deserves a comment - it's not normal for a search function to expect a match.